.github/workflows/component_features.yml

```
@@ -1,35 +1,65 @@
+# Component Features - Linux
+#
+# Validates that each component feature compiles
+#
+# Runs on:
+#  - scheduled UTC midnight Tues-Sat
+#  - on PR comment (see comment-trigger.yml)
+#  - on demand from github actions UI
+
 name: Component Features - Linux
 
 on:
   workflow_call:
+  workflow_dispatch:
+  schedule:
+    # At midnight UTC Tue-Sat
+    - cron: '0 0 * * 2-6'
 
 jobs:
-  check-component-features:
+  check-component-features-schedule:
+    # use free tier on the schedule
+    runs-on: ubuntu-latest
+    if: github.event_name == 'schedule'
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v3
+        with:
+          submodules: "recursive"
+
+      - run: sudo -E bash scripts/environment/bootstrap-ubuntu-20.04.sh
+      - run: bash scripts/environment/prepare.sh
+      - run: echo "::add-matcher::.github/matchers/rust.json"
+      - run: make check-component-features
+
+  check-component-features-comment:
+    # expedite results when running on demand
     runs-on: [linux, ubuntu-20.04-8core]
```

Could we use a ternary operator here to consolidate the jobs? It looks like this should work from reading the documentation but I'm not positive.
```suggestion
    runs-on: ${{ github.event_name == 'schedule' && 'ubuntu-latest' || fromJSON('["linux", "ubuntu-20.04-8core"]') }}
```

---

😮 I tried that in the playground and it worked, that's pretty awesome. I didn't realize that you could use expression syntax outside of `if` conditionals.

---

[e472e26](https://github.com/vectordotdev/vector/pull/17773/commits/e472e26ab457b79549359615826b78beaa70699c)